app/src/main/java/com/hover/stax/bounties/BountyEmailFragment.kt

```diff
@@ -0,0 +1,114 @@
+package com.hover.stax.bounties
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.NavHostFragment
+import com.hover.stax.R
+import com.hover.stax.databinding.FragmentBountyEmailBinding
+import com.hover.stax.navigation.NavigationInterface
+import com.hover.stax.utils.Utils.getString
+import com.hover.stax.utils.Utils.logAnalyticsEvent
+import com.hover.stax.utils.Utils.logErrorAndReportToFirebase
+import com.hover.stax.utils.Utils.saveString
+import com.hover.stax.utils.network.NetworkMonitor
+import com.hover.stax.views.AbstractStatefulInput
+import com.hover.stax.views.StaxDialog
+import com.hover.stax.views.StaxTextInputLayout
+import org.koin.androidx.viewmodel.ext.android.sharedViewModel
+import org.koin.java.KoinJavaComponent.get
+
+class BountyEmailFragment : Fragment(), NavigationInterface, View.OnClickListener {
+    private var emailInput: StaxTextInputLayout? = null
+    private var binding: FragmentBountyEmailBinding? = null
+    private var dialog: StaxDialog? = null
+    private val networkMonitor = get(NetworkMonitor::class.java)
+    private val viewModel: BountyViewModel by sharedViewModel()
+
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        binding = FragmentBountyEmailBinding.inflate(inflater, container, false)
+        return binding!!.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        observeEmailResult()
+        emailInput = binding!!.emailInput
+        emailInput!!.text = getString(BountyActivity.EMAIL_KEY, requireActivity())
+        binding!!.continueEmailBountyButton.setOnClickListener(this)
+    }
+
+    override fun onClick(v: View) {
+        if (networkMonitor.isNetworkConnected) {
+            logAnalyticsEvent(getString(R.string.clicked_bounty_email_continue_btn), requireContext())
+            if (validates()) {
+                emailInput!!.isEnabled = false
+                viewModel.uploadBountyUser(emailInput!!.text)
+                emailInput!!.setState(getString(R.string.bounty_uploading_email), AbstractStatefulInput.INFO)
+            } else {
+                emailInput!!.setState(getString(R.string.bounty_email_error), AbstractStatefulInput.ERROR)
+            }
+        } else {
+            showOfflineDialog()
+        }
+    }
+
+    private fun showOfflineDialog() {
+        StaxDialog(requireActivity())
+                .setDialogTitle(R.string.internet_required)
+                .setDialogMessage(R.string.internet_required_bounty_desc)
+                .setPosButton(R.string.btn_ok, null)
+                .makeSticky()
+                .showIt()
+    }
+
+    private fun showEdgeCaseErrorDialog() {
+        StaxDialog(requireActivity())
```

Comment:
Same goes for this one.

Reply:
Thanks, this is now fixed in recent push